app/src/lib/stores/helpers/infer-comparison-branch.ts

```diff
@@ -74,47 +80,6 @@ function getTargetBranchOfPullRequest(
   return findBranch(branches, pr.base.ref)
 }
 
-/**
- * For `inferComparisonBranch` case where inferring for a forked repository
- *
- * Returns the default branch of the fork if it's ahead of `currentBranch`.
- * Otherwise, the default branch of the parent is returned.
- */
-async function getDefaultBranchOfFork(
-  repository: Repository,
-  branches: ReadonlyArray<Branch>,
-  currentBranch: Branch,
-  getRemotes: RemotesGetter,
-  comparisonCache: ComparisonCache
-): Promise<Branch | null> {
-  // this is guaranteed to exist since this function
-  // is only called if the ghRepo is not null
-  const ghRepo = repository.gitHubRepository!
-  const defaultBranch = getDefaultBranchOfGitHubRepo(branches, ghRepo)
-  if (defaultBranch === null) {
-    return getMasterBranch(branches)
-  }
-
-  const aheadBehind = comparisonCache.get(
-    currentBranch.tip.sha,
-    defaultBranch.tip.sha
-  )
```

Comment:
This read from the cache was causing the inconsistencies: the cache is not always populated and that caused differences on the inferred branch depending on the actions that the user had taken before.